File: Assets/Scripts/CameraController.js
#pragma strict

var player: GameObject;
var offset: Vector3;
var lower: Vector3;

function Start () {
  lower.z = -5;  //shows less of behind the player object
  lower.y = 2.5; //gets camera closer to player object
  transform.position = transform.position + player.transform.position;  //adjusts camera for non-zero starting position on player
  offset = transform.position - player.transform.position - lower;  //creates the initial offset distance from player
}

function LateUpdate () {
  transform.position = player.transform.position + offset;
}

File: Assets/Scripts/CameraController.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public partial class CameraController : MonoBehaviour
{
    public GameObject player;
    public Vector3 offset;
    public Vector3 lower;
    public virtual void Start()
    {
        this.lower.z = -5; //shows less of behind the player object
        this.lower.y = 2.5f; //gets camera closer to player object
        this.transform.position = this.transform.position + this.player.transform.position; //adjusts camera for non-zero starting position on player
        this.offset = (this.transform.position - this.player.transform.position) - this.lower; //creates the initial offset distance from player
    }

    public virtual void LateUpdate()
    {
        this.transform.position = this.player.transform.position + this.offset;
    }

}